pallets/file-system/src/types.rs

```diff
@@ -118,3 +121,6 @@ pub type MaxMultiAddresses<T> =
 
 /// Alias for a bounded vector of [`MultiAddress`].
 pub type MultiAddresses<T> = BoundedVec<MultiAddress<T>, MaxMultiAddresses<T>>;
+
+/// Alias for the `Provider` type used in the ProvidersInterface.
+pub type ProviderFor<T> = <<T as crate::Config>::Providers as ProvidersInterface>::Provider;
```

Comment:
Just as a comment, in my latest PR I have updated all `Provider` references in the interfaces to be called `ProviderId` to be more clear on what the type actually represents, that's why I'm preeliminary suggesting naming changes to keep up with that. I can make those changes in my PR if needed